tests/unit/test1_unit_tests/src/main.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include "zcbor_decode.h"
+#include "zcbor_encode.h"
+#include "zcbor_debug.h"
+
+void test_int64(void)
+{
+	uint8_t payload[100] = {0};
+	int64_t int64;
+	int32_t int32;
+	zcbor_state_t state_d;
+	zcbor_state_t state_e;
+
+	zcbor_new_state(&state_e, 1, payload, sizeof(payload), 0);
+	zcbor_new_state(&state_d, 1, payload, sizeof(payload), 10);
+
+	zassert_true(zcbor_int64_put(&state_e, 5), NULL);
+	zassert_false(zcbor_int64_expect(&state_d, 4), NULL);
+	zassert_false(zcbor_int64_expect(&state_d, 6), NULL);
+	zassert_false(zcbor_int64_expect(&state_d, -5), NULL);
+	zassert_false(zcbor_int64_expect(&state_d, -6), NULL);
+	zassert_true(zcbor_int64_expect(&state_d, 5), NULL);
+
+	zassert_true(zcbor_int32_put(&state_e, 5), NULL);
+	zassert_true(zcbor_int64_expect(&state_d, 5), NULL);
+
+	zassert_true(zcbor_int64_put(&state_e, 5), NULL);
+	zassert_true(zcbor_int32_expect(&state_d, 5), NULL);
+
+	zassert_true(zcbor_int64_put(&state_e, 5000000000), NULL);
+	zassert_false(zcbor_int32_decode(&state_d, &int32), NULL);
+	zassert_true(zcbor_int64_decode(&state_d, &int64), NULL);
+	zassert_equal(int64, 5000000000, NULL);
+
+	zassert_true(zcbor_int64_encode(&state_e, &int64), NULL);
+	zassert_false(zcbor_int64_expect(&state_d, 5000000001), NULL);
+	zassert_true(zcbor_int64_expect(&state_d, 5000000000), NULL);
+
+	zassert_true(zcbor_int64_put(&state_e, 0x80000000), NULL);
+	zassert_false(zcbor_int32_decode(&state_d, &int32), NULL);
+	zassert_true(zcbor_uint32_expect(&state_d, 0x80000000), NULL);
+
+	zassert_true(zcbor_int32_put(&state_e, -505), NULL);
+	zassert_true(zcbor_int64_expect(&state_d, -505), NULL);
+
+	zassert_true(zcbor_int64_put(&state_e, -5000000000000), NULL);
+	zassert_false(zcbor_int64_expect(&state_d, -5000000000001), NULL);
+	zassert_false(zcbor_int64_expect(&state_d, -4999999999999), NULL);
+	zassert_false(zcbor_int64_expect(&state_d, 5000000000000), NULL);
+	zassert_false(zcbor_int64_expect(&state_d, 4999999999999), NULL);
+	zassert_true(zcbor_int64_expect(&state_d, -5000000000000), NULL);
+
+	zassert_true(zcbor_uint64_put(&state_e, 0x8000000000000000), NULL);
```

Comment:
Would it be more clear if we do something like this? Or maybe use the `INTxx_MAX`

```suggestion
	zassert_true(zcbor_uint64_put(&state_e, (2**64)+0x1000), NULL);
```